test.js

```
@@ -31,6 +36,11 @@ Upcase.prototype._transform = function(chunk, encoding, cb) {
     cb();
 };
 
+W.prototype._write = function () {
+    this.emit("error", new Error("Write failed"));
+    this.emit("error", new Error("Write failed"));
```

@MartinKolarik I am not convinced. Change the message of these errors so they are different. Then write an assertion for both.

I changed the message to "Write failed one" and "Write failed two" respectfully. To make the change pass I simply made (new) line 129 assert against "Write failed one".

---

Ok I think I understand what you meant now - after the first error is thrown, the second will be ignored - is that it? If yes, then this is indeed what I intended.

I noticed this change because one of the systems where we use this module started crashing rather frequently, after investigating I found that sometimes there are multiple error events emitted for one error (IIRC tar-fs was one of the libs doing that) and after the last update those were not handled.

I agree that swallowing errors is not great, but if the promise has already been rejected, the appropriate error handling code was executed, and you probably don't care about any subsequent errors. There might be cases where you'd prefer to have the error handler called multiple times but I don't see a way to do that with promises.

The current behavior makes this module almost unusable because you have no way to handle all errors reliably and prevent the whole process from crashing. Additionally, this change was made in a minor version release, which is supposed to be backward compatible.